app/src/main/java/com/openlattice/chronicle/services/upload/UploadWorker.kt

```diff
@@ -124,16 +123,23 @@ class UploadWorker(context: Context, params: WorkerParameters) : Worker(context,
             val w = Stopwatch.createStarted()
             val data = nextEntries
                 .map { qe -> qe.data }
-                .map { qe -> JsonSerializer.deserializeQueueEntry(qe) }
-                .map { qe -> mapToModel(qe)}
+                .map { qe ->
+                    //Attempt to deserialize as legacy queue entry on exception.
+                    try {
+                        JsonSerializer.deserializeQueueEntry(qe)
+                    } catch (ex: Exception) {
+                        mapLegacyQueueEntry(JsonSerializer.deserializeLegacyQueueEntry(qe))
+                    }
```

Comment:
Exception thrown by `JsonSerializer.deserializeQueueEntry(qe)` isn't caught here, therefore ` mapLegacyQueueEntry(JsonSerializer.deserializeLegacyQueueEntry(qe))` is never called.

We should change the definition of `JsonSerializer.deserializeQueueEntry(qe)` to throw instead.

```
public static List<ChronicleDataUpload> deserializeQueueEntry(byte[] bytes) throws IOException {
    return mapper.readValue(bytes, new TypeReference<List<ChronicleDataUpload>>() {});
}
```

Reply:
Good catch! Updated.